sql/src/main/java/io/crate/execution/engine/collect/stats/JobsLogService.java

```diff
@@ -251,9 +252,8 @@ private void updateJobSink(int size, TimeValue expiration) {
             sink
         );
         LogSink<JobContextLog> oldSink = jobsLogSink;
-        newSink.addAll(oldSink);
-        jobsLogSink = newSink;
         jobsLogs.updateJobsLog(newSink);
+        jobsLogSink = newSink;
```

Comment:
I think this assignment should also be protected within the lock logic.
If you exit the updateJobsLog() method the write lock is released and pending threads could still write to the old sink. No?
(same for the operations).

Reply:
no this reference is only used to proper close the sink if replaced.

Reply:
Yep, thx! I was confused. Moving the closing to JobsLogs improves the readability ;-)